src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -48,7 +48,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NativeLibrary Condition="'$(IlcMultiModule)' == 'true'" Include="$(SharedLibrary)" />
       <NativeLibrary Condition="$(NativeLib) == ''" Include="$(IlcSdkPath)libbootstrapper.a" />
       <NativeLibrary Condition="$(NativeLib) != ''" Include="$(IlcSdkPath)libbootstrapperdll.a" />
-      <NativeLibrary Include="$(IlcSdkPath)$(FullRuntimeName).a" />
+      <NativeLibrary Include="$(IlcSdkPath)$(FullRuntimeName).a;$(IlcSdkPath)libstdc++compat.a" />
```

Comment:
Could you please add property for this, something like `LinkStandardCPlusPlusLibrary`. When it is not set, use the shim. When it is set, skip the shim and use `-lstdc++` instead.

Also, set this property in one of the smoke tests so that we know that it works.

Reply:
Any particular reason for naming the property `SkipCppCompat`?

I prefer for the public surface to be explainable in terms of other public surface and not in terms of internal implementation details. "This enables linking with stdc++ library" is much easier to explain than "This disables internal implementation detail that may cause problems when linking with stdc++ library".

Also, could you please also set it in one of the smoke tests, for example here: https://github.com/dotnet/runtime/blob/main/src/tests/nativeaot/SmokeTests/Preinitialization/Preinitialization.csproj#L13 so that we have validation that it works?

Reply:
> Any particular reason for naming the property SkipCppCompat?

I was already working and testing on this infra before I saw your comment. I'm still working on linux, will push updates once linux is fixed.